### PR TITLE
Finished Inerfaces menu library

### PR DESCRIPTION
**Pull request by Ronenii**
Finished the interfaces menu.

We now need to organize code and combine both of our menus into a single menu

### CODE REVIEW

File: Ex04.Menus.Test/Program.cs [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
+            ActionMenuItem actionItemShowTime = new ActionMenuItem("Show Time", subItemShowDateTime, new ShowTime());
+            subItemShowDateTime.AddMenuItem(actionItemShowDate);
+            subItemShowDateTime.AddMenuItem(actionItemShowTime);
+
+            // Init main menu option 2 (sub menu of show version and count spaces)
+            SubMenuItem subItemVersionSpaces = new SubMenuItem("Version and Spaces", mainMenu, v_IsNotMainMenu);
+            ActionMenuItem actionItemShowVersion = new ActionMenuItem("Show Version", subItemVersionSpaces, new ShowVersion());
+            ActionMenuItem actionItemCountSpaces = new ActionMenuItem("Count Spaces", subItemVersionSpaces, new CountSpaces());
+            subItemVersionSpaces.AddMenuItem(actionItemShowVersion);
+            subItemVersionSpaces.AddMenuItem(actionItemCountSpaces);
```
Comment (idobi111) [inline Ex04.Menus.Test/Program.cs]: extract to method like init menu
in my part I create new class of menu handler and create all of this logic as init (look there let me know what do u think :)

File: Ex04.Menus.Test/Program.cs [hunk 2]
```diff
@@ -1,12 +1,38 @@
 ﻿using System;
+using B23_Ex04_Ronen_319047718_Ido_315942193;
 
 namespace Ex04.Menus.Test
 {
     public class Program
     {
         public static void Main()
         {
+            const bool v_IsMainMenu = true;
```
Comment (idobi111) [inline Ex04.Menus.Test/Program.cs:10]: only one filed is relevant
if its not main so it clearly no Main

File: Ex04.Menus.Test/InterfaceFuncs.cs [hunk 3]
```diff
@@ -10,31 +10,57 @@ public class ShowDate : IMenuObserver
...
+            string sentence = Console.ReadLine();
+            int spaceCounter = 0;
+
+            foreach(char c in sentence)
+            {
+                if(c.Equals(' '))
+                {
+                    spaceCounter++;
+                }
+            }
```
Comment (idobi111) [inline Ex04.Menus.Test/InterfaceFuncs.cs]:
```suggestion
                   string sentence = Console.ReadLine();
            int spacesCount = sentence.Split(' ').Length - 1;
```

Comment (idobi111) [reply Ex04.Menus.Test/InterfaceFuncs.cs]: what do u think?:

**Review by idobi111 — CHANGES_REQUESTED** [review]

File: Ex04.Menus.Test/Program.cs (hunk 1 above)
Comment (Ronenii) [reply Ex04.Menus.Test/Program.cs]: This is just a placeholder until we implement the menuHandler

File: Ex04.Menus.Test/Program.cs (hunk 2 above)
Comment (Ronenii) [reply Ex04.Menus.Test/Program.cs:10]: Got it no problem

File: Ex04.Menus.Test/InterfaceFuncs.cs (hunk 3 above)
Comment (Ronenii) [reply Ex04.Menus.Test/InterfaceFuncs.cs]: Nice

File: Ex04.Menus.Test/InterfaceFuncs.cs [hunk 4]
```diff
@@ -10,31 +10,57 @@ public class ShowDate : IMenuObserver
...
     }
 
     public class ShowTime : IMenuObserver
     {
         public void Execute()
         {
-            throw new NotImplementedException();
+            DateTime date = DateTime.Now;
+            Console.Clear();
+            Console.WriteLine("Current Date: " + date.ToString("HH:mm:ss"));
```
Comment (idobi111) [inline Ex04.Menus.Test/InterfaceFuncs.cs]: Current time?

**Review by idobi111 — CHANGES_REQUESTED** [review]

File: Ex04.Menus. Interfaces/MenuItem.cs [hunk 5]
```diff
@@ -5,36 +5,62 @@
...
         public MenuItem(string i_Title, MenuItem i_Parent)
         {
-            m_Title = i_Title;
-            m_Parent = i_Parent;
-            m_Path += string.Format("{0}->{1}", i_Parent.Path, i_Parent.Title );
+            r_Title = i_Title;
+            r_Parent = i_Parent;
+
+            // If the parent is nul lthis means we are in the main menu and it still doesn't have a relative path
+            if (i_Parent != null)
```
Comment (idobi111) [inline Ex04.Menus. Interfaces/MenuItem.cs:44]:
```suggestion
            if (isMainMenu)
```

**Review by idobi111 — APPROVED** [review]

**Pull request merged**